src/devices/machine/mm5740.cpp

```diff
@@ -0,0 +1,566 @@
+	0b00000001111,  // X1 Y8
+	0b00100110111,  // X1 Y9
+	0b00100010111,  // X1 Y10
+	0b00100100111,  // X2 Y1
+	0b00100000111,  // X2 Y2
+	0b00000110111,  // X2 Y3
+	0b00000010111,  // X2 Y4
+	0b00000100111,  // X2 Y5
+	0b00000000111,  // X2 Y6
+	0b00100111011,  // X2 Y7
+	0b00100011011,  // X2 Y8
+	0b00100101011,  // X2 Y9
+	0b00100001011,  // X2 Y10
+	0b00000111011,  // X3 Y1
+	0b00000011011,  // X3 Y2
+	0b00000101011,  // X3 Y3
+	0b00000001011,  // X3 Y4
+	0b00100110011,  // X3 Y5
+	0b00100010011,  // X3 Y6
+	0b00100100011,  // X3 Y7
+	0b00100000011,  // X3 Y8
+	0b00000110011,  // X3 Y9
+	0b00000010011,  // X3 Y10
+	0b00000100011,  // X4 Y1
+	0b00000000011,  // X4 Y2
+	0b00100111101,  // X4 Y3
+	0b00100011101,  // X4 Y4
+	0b00100101101,  // X4 Y5
+	0b00100001101,  // X4 Y6
+	0b00000111101,  // X4 Y7
+	0b00000011101,  // X4 Y8
+	0b00000101101,  // X4 Y9
+	0b00000001101,  // X4 Y10
+	0b00100110101,  // X5 Y1
+	0b00100010101,  // X5 Y2
+	0b00100100101,  // X5 Y3
+	0b00100000101,  // X5 Y4
+	0b00000110101,  // X5 Y5
+	0b00000010101,  // X5 Y6
+	0b00000100101,  // X5 Y7
+	0b00000000101,  // X5 Y8
+	0b00100111001,  // X5 Y9
+	0b00100011001,  // X5 Y10
+	0b00100101001,  // X6 Y1
+	0b00100001001,  // X6 Y2
+	0b00000111001,  // X6 Y3
+	0b00000011001,  // X6 Y4
+	0b00000101001,  // X6 Y5
+	0b00000001001,  // X6 Y6
+	0b00100110001,  // X6 Y7
+	0b00100010001,  // X6 Y8
+	0b00100100001,  // X6 Y9
+	0b00100000001,  // X6 Y10
+	0b00000110001,  // X7 Y1
+	0b00000010001,  // X7 Y2
+	0b00000100001,  // X7 Y3
+	0b00000000001,  // X7 Y4
+	0b00100111110,  // X7 Y5
+	0b00100011110,  // X7 Y6
+	0b00100101110,  // X7 Y7
+	0b00100001110,  // X7 Y8
+	0b00000111110,  // X7 Y9
+	0b00000011110,  // X7 Y10
+	0b00000101110,  // X8 Y1
+	0b00000001110,  // X8 Y2
+	0b00100110110,  // X8 Y3
+	0b00100010110,  // X8 Y4
+	0b00100100110,  // X8 Y5
+	0b00100000110,  // X8 Y6
+	0b00000110110,  // X8 Y7
+	0b00000010110,  // X8 Y8
+	0b00000100110,  // X8 Y9
+	0b00000000110,  // X8 Y10
+	0b00100111010,  // X9 Y1
+	0b00100011010,  // X9 Y2
+	0b00100101010,  // X9 Y3
+	0b00100001010,  // X9 Y4
+	0b00000111010,  // X9 Y5
+	0b00000011010,  // X9 Y6
+	0b00000101010,  // X9 Y7
+	0b00000001010,  // X9 Y8
+	0b00100110010,  // X9 Y9
+	0b00100010010,  // X9 Y10
+};
+
+
+//-------------------------------------------------
+//  device_start - device-specific startup
+//-------------------------------------------------
+
+void mm5740_device::device_start()
+{
+	// resolve callbacks
+	m_read_x1.resolve_safe(0x3ff);
+	m_read_x2.resolve_safe(0x3ff);
+	m_read_x3.resolve_safe(0x3ff);
+	m_read_x4.resolve_safe(0x3ff);
+	m_read_x5.resolve_safe(0x3ff);
+	m_read_x6.resolve_safe(0x3ff);
+	m_read_x7.resolve_safe(0x3ff);
+	m_read_x8.resolve_safe(0x3ff);
+	m_read_x9.resolve_safe(0x3ff);
+	m_read_shift.resolve_safe(0);
+	m_read_control.resolve_safe(0);
+	m_write_data_ready.resolve_safe();
+
+	// allocate timers
+	m_scan_timer = timer_alloc();
+	m_scan_timer->adjust(attotime::from_hz(60), 0, attotime::from_hz(60));
```

Comment:
Where does the magic frequency come from?  I'm pretty sure it's not generated internally.  You should either have a write line member for the clock input which the driver can attach to a clock device or something, or instantiate your own clock device with an MCFG fragment.